documentation/jetty-documentation/src/main/asciidoc/programming-guide/migration/migration.adoc

```diff
@@ -0,0 +1,130 @@
+
+==== Required Java Version Changes
+
+[cols="1,1", options="header"]
+|===
+| Jetty 9.4.x | Jetty 10.0.x
+| Java 8 | Java 11
+|===
+
+=== WebSocket Migration Guide
+
+Migrating from Jetty 9.4.x to Jetty 10.0.x requires changes in the coordinates of the Maven artifact dependencies for WebSocket. Some of these classes have also changed name and package. This is not a comprehensive list of changes but should cover the most common changes encountered during migration.
+
+==== Maven Artifacts Changes
+
+[cols="1a,1a", options="header"]
+|===
+| Jetty 9.4.x | Jetty 10.0.x
+
+| `org.eclipse.jetty.websocket:**websocket-api**`
+| `org.eclipse.jetty.websocket:**websocket-jetty-api**`
+
+| `org.eclipse.jetty.websocket:**websocket-server**`
+| `org.eclipse.jetty.websocket:**websocket-jetty-server**`
+
+| `org.eclipse.jetty.websocket:**websocket-client**`
+| `org.eclipse.jetty.websocket:**websocket-jetty-client**`
+
+| `org.eclipse.jetty.websocket:**javax-websocket-server-impl**`
+| `org.eclipse.jetty.websocket:**websocket-javax-server**`
+
+| `org.eclipse.jetty.websocket:**javax-websocket-client-impl**`
+| `org.eclipse.jetty.websocket:**websocket-javax-client**`
+
+|===
+
+==== Class Names Changes
+
+[cols="1a,1a", options="header"]
+|===
+| Jetty 9.4.x | Jetty 10.0.x
+
+| `org.eclipse.jetty.websocket.**server.NativeWebSocketServletContainerInitializer**`
+| `org.eclipse.jetty.websocket.**server.config.JettyWebSocketServletContainerInitializer**`
+
+| `org.eclipse.jetty.websocket.**jsr356.server.deploy.WebSocketServerContainerInitializer**`
+| `org.eclipse.jetty.websocket.**javax.server.config.JavaxWebSocketServletContainerInitializer**`
+
+| `org.eclipse.jetty.websocket.**servlet.WebSocketCreator**`
+| `org.eclipse.jetty.websocket.**server.JettyWebSocketCreator**`
+
+| `org.eclipse.jetty.websocket.**servlet.ServletUpgradeRequest**`
+| `org.eclipse.jetty.websocket.**server.JettyServerUpgradeRequest**`
+
+| `org.eclipse.jetty.websocket.**servlet.ServletUpgradeResponse**`
+| `org.eclipse.jetty.websocket.**server.JettyServerUpgradeResponse**`
+
+| `org.eclipse.jetty.websocket.**servlet.WebSocketServlet**`
+| `org.eclipse.jetty.websocket.**server.JettyWebSocketServlet**`
+
+| `org.eclipse.jetty.websocket.**servlet.WebSocketServletFactory**`
+| `org.eclipse.jetty.websocket.**server.JettyWebSocketServletFactory**`
+|===
+
+==== Example Code
+
+[cols="1a,1a", options="header"]
+|===
+| Jetty 9.4.x
+| Jetty 10.0.x
+
+|
+[source,java]
+----
+public class ExampleWebSocketServlet extends WebSocketServlet
+{
+    @Override
+    public void configure(WebSocketServletFactory factory)
+    {
+        factory.setCreator(new WebSocketCreator()
+        {
+            @Override
+            public Object createWebSocket(ServletUpgradeRequest req, ServletUpgradeResponse resp)
+            {
+                return new ExampleEndpoint();
+            }
+        });
+    }
+}
+----
+
+|
+[source,java]
+----
+public class ExampleWebSocketServlet extends JettyWebSocketServlet
```

Comment:
Is it possible to highlight the `Jetty` prefix somehow? Above you were surrounding with `**`.

Reply:
I tried this and it didn't work because it is interpreting the contents as java.